src/object/class_object.c

```diff
@@ -4069,6 +4069,33 @@ classobj_find_constraint_by_attrs (SM_CLASS_CONSTRAINT * cons_list, DB_CONSTRAIN
   const char **namep;
   int i, len, order;
 
+  /* for foreign key, need to check redundancy first */
+  if (new_cons == SM_CONSTRAINT_FOREIGN_KEY)
+    {
+      for (cons = cons_list; cons; cons = cons->next)
```

Comment:
I suggest another way.

If SM_CONSTRAINT_PRIMARY_KEY is found in classobj_find_constraint_by_attrs() function and it is not returned immediately, SM_CONSTRAINT_FOREIGN_KEY can be found.

If the existing_con returned by the classobj_find_constraint_by_name() function is SM_CONSTRAINT_FOREIGN_KEY, the SM_CONSTRAINT_FOREIGN_KEY can be checked in the classobj_check_index_compatibility() function.

If SM_CONSTRAINT_FOREIGN_KEY already exists, SM_SHARE_INDEX is returned instead of SM_CREATE_NEW_INDEX.

For example, how about returning SM_NOT_SHARE_INDEX_AND_WARNING instead of SM_SHARE_INDEX to raise an error?